src/iidm/util/TwtData.cpp

```diff
@@ -0,0 +1,610 @@
+    TwtData(twt, 0, 0, epsilonX, applyReactanceCorrection, false) {
+}
+
+TwtData::TwtData(const ThreeWindingsTransformer& twt, double epsilonX, bool applyReactanceCorrection, bool twtSplitShuntAdmittance) :
+    TwtData(twt, 0, 0, epsilonX, applyReactanceCorrection, twtSplitShuntAdmittance) {
+}
+
+TwtData::TwtData(const ThreeWindingsTransformer& twt, int phaseAngleClock2, int phaseAngleClock3, double epsilonX, bool applyReactanceCorrection, 
+                 bool twtSplitShuntAdmittance) :
+    m_id(twt.getId()),
+    m_p1(twt.getLeg1().getTerminal().getP()),
+    m_q1(twt.getLeg1().getTerminal().getQ()),
+    m_p2(twt.getLeg2().getTerminal().getP()),
+    m_q2(twt.getLeg2().getTerminal().getQ()),
+    m_p3(twt.getLeg3().getTerminal().getP()),
+    m_q3(twt.getLeg3().getTerminal().getQ()),
+    m_u1(getV(twt.getLeg1())),
+    m_theta1(getTheta(twt.getLeg1())),
+    m_u2(getV(twt.getLeg2())),
+    m_theta2(getTheta(twt.getLeg2())),
+    m_u3(getV(twt.getLeg3())),
+    m_theta3(getTheta(twt.getLeg3())),
+    m_r1(getR(twt.getLeg1())),
+    m_x1(LinkData::getFixedX(getX(twt.getLeg1()), epsilonX, applyReactanceCorrection)),
+    m_r2(getR(twt.getLeg2())),
+    m_x2(LinkData::getFixedX(getX(twt.getLeg2()), epsilonX, applyReactanceCorrection)),
+    m_r3(getR(twt.getLeg3())),
+    m_x3(LinkData::getFixedX(getX(twt.getLeg3()), epsilonX, applyReactanceCorrection)),
+    m_g11(getG1(twt.getLeg1(), twtSplitShuntAdmittance)),
+    m_b11(getB1(twt.getLeg1(), twtSplitShuntAdmittance)),
+    m_g12(getG2(twt.getLeg1(), twtSplitShuntAdmittance)),
+    m_b12(getB2(twt.getLeg1(), twtSplitShuntAdmittance)),
+    m_g21(getG1(twt.getLeg2(), twtSplitShuntAdmittance)),
+    m_b21(getB1(twt.getLeg2(), twtSplitShuntAdmittance)),
+    m_g22(getG2(twt.getLeg2(), twtSplitShuntAdmittance)),
+    m_b22(getB2(twt.getLeg2(), twtSplitShuntAdmittance)),
+    m_g31(getG1(twt.getLeg3(), twtSplitShuntAdmittance)),
+    m_b31(getB1(twt.getLeg3(), twtSplitShuntAdmittance)),
+    m_g32(getG2(twt.getLeg3(), twtSplitShuntAdmittance)),
+    m_b32(getB2(twt.getLeg3(), twtSplitShuntAdmittance)),
+    m_alpha1(alpha(twt.getLeg1())),
+    m_alpha2(alpha(twt.getLeg2())),
+    m_alpha3(alpha(twt.getLeg3())),
+    m_ratedU1(twt.getLeg1().getRatedU()),
+    m_ratedU2(twt.getLeg2().getRatedU()),
+    m_ratedU3(twt.getLeg3().getRatedU()),
+    m_connected1(twt.getLeg1().getTerminal().isConnected()),
+    m_connected2(twt.getLeg2().getTerminal().isConnected()),
+    m_connected3(twt.getLeg3().getTerminal().isConnected()),
+    m_mainComponent1(isMainComponent(twt.getLeg1())),
+    m_mainComponent2(isMainComponent(twt.getLeg2())),
+    m_mainComponent3(isMainComponent(twt.getLeg3())),
+    m_phaseAngleClock2(phaseAngleClock2),
+    m_phaseAngleClock3(phaseAngleClock3),
+    m_ratedU0(twt.getRatedU0()) {
+    m_rho1 = rho(twt.getLeg1(), m_ratedU0);
+    m_rho2 = rho(twt.getLeg2(), m_ratedU0);
+    m_rho3 = rho(twt.getLeg3(), m_ratedU0);
```

Comment:
Because `m_ratedU0` need to be initialized, and we cannot initialize `m_rho1` in initializer list before `m_ratedU0` without having a compiler warning. The solution could be to reorder attributes so that `m_rho{1,2,3}` are declared after `m_ratedU0` ?

Reply:
No need to use m_rated0 because it’s equivalent to twt.getRated0()

Reply:
Yes it is needed because it is returned by `TwtData::getRatedU0()`